orchestrator/api/helpers.py

```diff
@@ -73,86 +73,82 @@ def _add_sort_to_query(query: Query, sort: Optional[List[str]]) -> Query:
     return query
 
 
-def _query_with_filters(  # noqa: C901
-    response: Response,
-    query: Query,
-    range: Optional[List[int]] = None,
+def query_with_filters(  # noqa: C901
+    stmt: Select,
     sort: Optional[List[str]] = None,
     filters: Optional[List[str]] = None,
-) -> List:
+) -> Select:
     if filters is not None:
```

Comment:
In the future we may want to decomplect this code a bit. Huge for loop with nested ifs...

Reply:
Yeah, it's already better than a month ago, but can definitely be simplified further